### PR TITLE
[Enterprise Search] Add update sync job error endpoint

### PR DESCRIPTION
**Pull request by timgrein**
Adds the update connector sync job error endpoint. This also sets the `status` of the corresponding sync job to `error`;

### CODE REVIEW

**Comment by timgrein** [comment]
@elasticsearchmachine run elasticsearch-ci

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java [hunk 1]
```diff
@@ -272,7 +276,8 @@ public List<RestHandler> getRestHandlers(
                     new RestPostConnectorSyncJobAction(),
                     new RestDeleteConnectorSyncJobAction(),
                     new RestCancelConnectorSyncJobAction(),
-                    new RestCheckInConnectorSyncJobAction()
+                    new RestCheckInConnectorSyncJobAction(),
```
Comment (jedrazb) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java:279]: Nit for dealing with merge conflicts, let's try to keep it ordered alphabetically Cancel < Check < Delete. WYT?

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java [hunk 2]
```diff
@@ -264,6 +264,43 @@ public void onFailure(Exception e) {
...
+     * @param listener               The action listener to invoke on response/failure.
+     */
+    public void updateConnectorSyncJobError(String connectorSyncJobId, String error, ActionListener<UpdateResponse> listener) {
+        final UpdateRequest updateRequest = new UpdateRequest(CONNECTOR_SYNC_JOB_INDEX_NAME, connectorSyncJobId).setRefreshPolicy(
+            WriteRequest.RefreshPolicy.IMMEDIATE
+        )
+            .doc(
+                Map.of(
+                    ConnectorSyncJob.ERROR_FIELD.getPreferredName(),
+                    error,
```
Comment (jedrazb) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java:282]: What happens if we want to reset the error to `null`? As a sideeffect we would set the sync job to error (not sure if it's possible with sync job, setting error to null for connector is possible)

Followup: let's discuss the allowed state transitions

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/UpdateConnectorSyncJobErrorAction.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            ActionRequestValidationException validationException = null;
+
+            if (Strings.isNullOrEmpty(connectorSyncJobId)) {
+                validationException = addValidationError(
+                    ConnectorSyncJobConstants.EMPTY_CONNECTOR_SYNC_JOB_ID_ERROR_MESSAGE,
+                    validationException
+                );
+            }
+
+            if (Strings.isNullOrEmpty(error)) {
```
Comment (jedrazb) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/UpdateConnectorSyncJobErrorAction.java:95]: We can't reset the error to null?

**Review by jedrazb — APPROVED** [review]
LGTM, left some questions related to allowed/supported state trasitions

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java (hunk 1 above)
Comment (timgrein) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java:279]: Fair point, actually I thought I've ordered it in a different PR, but probably forgot it here.

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java (hunk 2 above)
Comment (timgrein) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java:282]: I'm actually not sure, if you should be able to "reset" a sync job. Currently, if a sync job fails it stops and cannot be picked up again as we don't do checkpointing (we've incremental syncs, but for a different use case).

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/UpdateConnectorSyncJobErrorAction.java (hunk 3 above)
Comment (timgrein) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/UpdateConnectorSyncJobErrorAction.java:95]: See comment above: I don't think we currently reset sync job errors.

**Pull request merged**